### PR TITLE
Introduces Stack Layout

### PR DESCRIPTION
**Pull request by andresgalante**
@siddharthkp @landitus I really need your help with this one:

As you know `StackLayout` is a subset of flex and replaces `Stack`. Can you play with it, maybe create some stories and see how it behaves, and write your thoughts.

This I am looking to know are:
- Are we missing any option on the props? for example, distribution `space-between`
- Can you think of things you cannot build with this?
- Is the `space` prop clear enough after you use it? can you think of a better word?

@siddharthkp I'll also need your help to validate the `space` string. It should only accept numbers from 1 to 10, or the words `none` or `auto`.

Docs are missing, I am working on it today.

Thanks 🙏

### CODE REVIEW

File: core/components/molecules/_stack-layout/stack-layout.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+You can nest StackLayouts or combine it with other layouts to create complex arrangements.
+
+### Do not use it for:
+
+- **Page chrome:** Use the App Frame instead for the page chrome.
+- **Create columns:** To create a column layout, use the Column Layout instead.
+- **Card List**: Use the Gallery layout instead.
+
+## Accecibility
```
Comment (francocorreasosa) [inline core/components/molecules/_stack-layout/stack-layout.md]:
```suggestion
## Accessibility
```

**Comment by siddharthkp** [comment]
Read the docs, this boy needs a lot more example to understand what the props do

**Comment by andresgalante** [comment]
@siddharthkp yeap, it does. Fer suggested creating a new issue for creating examples for the stacklayout since there are so many.

**Comment by andresgalante** [comment]
@siddharthkp @francocorreasosa I've opened #1416 and #1417 to track the rest of the work related to the stack layout, can we move this forward?

**Comment by francocorreasosa** [comment]
Let's merge this (after at least one more approval), then let's write the docs 🤓

**Pull request merged**